src/OpenTracing/Util/GlobalTracer.cs

```diff
@@ -0,0 +1,125 @@
+    /// often reasonable and cleaner for all of the usual DI reasons.
+    /// <para/>
+    /// That said, instrumentation for packages that are themselves statically
+    /// configured (e.g., JDBC drivers) may be unable to make use of said DI
+    /// mechanisms for <see cref="ITracer"/> access, and as such they should fall back on
+    /// <see cref="GlobalTracer"/>. By and large, OpenTracing instrumentation should
+    /// always allow the programmer to specify a <see cref="ITracer"/> instance to use for
+    /// instrumentation, though the <see cref="GlobalTracer"/> is a reasonable fallback or
+    /// default value.
+    /// </summary>
+    public sealed class GlobalTracer : ITracer
+    {
+        private static readonly object s_lock = new object();
+
+        /// <summary>
+        /// Singleton instance.
+        /// <para/>
+        /// Since we cannot prevent people using <see cref="Instance"/> as a constant,
+        /// this guarantees that references obtained before, during or after initialization
+        /// all behave as if obtained <em>after</em> initialization once properly initialized.<br/>
+        /// As a minor additional benefit it makes it harder to circumvent the <see cref="ITracer"/> API.
+        /// </summary>
+        private static readonly GlobalTracer s_instance = new GlobalTracer();
+
+        /// <summary>
+        /// Returns the constant <see cref="GlobalTracer"/>.
+        /// <para/>
+        /// All methods are forwarded to the currently configured tracer.<br/>
+        /// Until a tracer is explicitly configured via <see cref="Register(ITracer)"/>,
+        /// the <see cref="NoopTracer"/> is used.
+        /// </summary>
+        /// <returns>The global tracer constant.</returns>
+        /// <seealso cref="Register(ITracer)"/>
+        public static ITracer Instance => s_instance;
+
+        /// <summary>
+        /// Identify whether a <see cref="ITracer"/> has previously been registered.
+        /// <para/>
+        /// This check is useful in scenarios where more than one component may be responsible
+        /// for registering a tracer.
+        /// </summary>
+        /// <returns>Whether a tracer has been registered.</returns>
+        public static bool IsRegistered()
+        {
+            return !(s_instance._tracer is NoopTracer);
+        }
+
+        /// <summary>
+        /// Register a <see cref="ITracer"/> to back the behaviour of the global tracer (<see cref="Instance"/>).
+        /// <para/>
+        /// Registration is a one-time operation, attempting to call it more often will result in a runtime exception.
+        /// <para/>
+        /// Every application intending to use the global tracer is responsible for registering it once
+        /// during its initialization.
+        /// </summary>
+        /// <param name="tracer">Tracer to use as global tracer.</param>
+        public static void Register(ITracer tracer)
```

Comment:
It's possible a developer at the composition layer wants to register one tracer but someone wants to wrap that for composition (or they have a InitializeStandardTracer in a base class and then some custom wrapping they want). The current code prevents that.
A possible way to handle it is

```C#
public delegate ITracer ExchangeTracerDelegate(ITracer existingTracer);

public static ITracer ExchangeTracer([NotNull] ExchangeTracerDelegate newTracerDelegate) // Oops, my love for nullability annotations from ReSharper is showing :)
{
    lock (s_lock)
    {
        s_instance._tracer = newTracerDelegate(s_instance._tracer);
    }
}
```

Reply:
Note we also do not support someone saying "I want to intercept trace messages from my callee here and down", which would require AsyncLocal for the tracer.
While I'd like to support that, I wouldn't do so until I had measured the performance of AsyncLocal.

Reply:
These are requirements that are not yet formally specified (https://github.com/opentracing/specification/issues/9 [multiple tracers], https://github.com/opentracing/specification/issues/76 [observers]) so they are no-goals of this implementation.

Reply:
@ndrwrbgs That's an interesting feature. I'd personally recommend you post an issue against the Java repo and have people discuss it ;)